tools/integration_tests/write_large_files/concurrent_write_files_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package write_large_files
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"sync"
+	"syscall"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+const (
+	FileOne                                    = "fileOne.txt"
+	FileTwo                                    = "fileTwo.txt"
+	FileThree                                  = "fileThree.txt"
+	NumberOfFilesInLocalDiskForConcurrentWrite = 3
+	DirForConcurrentWrite                      = "dirForConcurrentWrite"
+)
+
+func writeFile(fileName string, fileSize int64, wg *sync.WaitGroup) (err error) {
+	// Reduce thread count when it is done.
+	defer wg.Done()
+
+	filePath := path.Join(setup.MntDir(), DirForConcurrentWrite, fileName)
+	f, err := os.OpenFile(filePath, os.O_WRONLY|os.O_CREATE|syscall.O_DIRECT, setup.FilePermission_0600)
+	if err != nil {
+		err = fmt.Errorf("Open file for write at start: %v", err)
+		return
+	}
+
+	// Closing file at the end.
+	defer operations.CloseFile(f)
+
+	err = WriteChunkSizeInFile(f, filePath, int(fileSize), 0)
+	if err != nil {
+		err = fmt.Errorf("Error:%v", err)
+		return
+	}
+
+	filePathInGcsBucket := path.Join(setup.TestBucket(), DirForConcurrentWrite, fileName)
+	localFilePath := path.Join(TmpDir, fileName)
+	err = compareFileFromGCSBucketAndMntDir(filePathInGcsBucket, filePath, localFilePath)
+	if err != nil {
+		err = fmt.Errorf("Error:%v", err)
+		return
+	}
+
+	return
+}
+
+func TestMultipleFilesAtSameTime(t *testing.T) {
+	concurrentWriteDir := path.Join(setup.MntDir(), DirForConcurrentWrite)
+	err := os.Mkdir(concurrentWriteDir, setup.FilePermission_0600)
+	if err != nil {
+		t.Fatalf("Error in creating directory:%v", err)
+	}
+
+	// Clean up.
+	defer operations.RemoveDir(concurrentWriteDir)
+
+	// For waiting on threads.
+	var wg sync.WaitGroup
+
+	files := [NumberOfFilesInLocalDiskForConcurrentWrite]string{FileOne, FileTwo, FileThree}
```

Comment:
NumberOfFilesInLocalDiskForConcurrentWrite isn't required.

```
files := []string{FileOne, FileTwo, FileThree}
```

For looping, you can do the following:

```
for i, file:= range files {
```

or 

```
for i:= range files {
```

or 

```
for i:=0, i< len(files); i++ {
```